### PR TITLE
#182 - Enable Multi-column sorting in DataTable

### LINKED ISSUE #182: Enable Multi-column sorting in DataTable
## Basic information

* Tiller version: 1.7.4
* Module: data-display

## Enhancement description

Improve DataTable's sorting capabilities by allowing users to sort by multiple columns simultaneously.

### Current behaviour

The current implementation is strong for single-column sorting, and this enhancement seeks to make multi-column sorting more robust and intuitive.

### Wanted behaviour

Enable DataTable to sort by multiple columns simultaneously.

### PR DESCRIPTION
**Pull request by ivam5**
## Basic information

* Tiller version: 1.7.4
* Module: data-display

## Description
Enable multi-column sorting functionality in the DataTable component.

### Related issue

Closes #182 

## Types of changes

- Enhancement (non-breaking change which enhances existing functionality)
- New feature (non-breaking change which adds functionality)
- Docs change

## Checklist

- [x] I have read the project's **CONTRIBUTING** document
- [x] I have checked my code with the project's static analysis tooling
- [x] I have formatted my code with the project's Prettier code-style configuration
- [x] I have checked my code for misspellings
- [x] I have organized my changes in easy-to-follow commits
- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly
- [x] I have added a story to cover my changes
- [x] All new and existing story tests pass

### CODE REVIEW

File: storybook/src/data-display/DataTable.mdx [hunk 1]
```diff
@@ -111,6 +111,67 @@ Here is an example utilizing the `useMemo` hook. This specific logic is applied
...
+```tsx
+<DataTable.Column header="Name" accessor="name" canSort={true} />
+<DataTable.Column header="Surname" accessor="surname" canSort={true} />
+```
+
+**2.** For multi-column sorting, apply sorting logic based on the priority of columns specified in the sort state:
+```tsx
+  const [dataTableState, dataTableHook] = useDataTable({
+    defaultSortBy: [
+      {
```
Comment (fbeceic) [inline storybook/src/data-display/DataTable.mdx:131]: Multiple columns should be defined here, right?

Comment (ivam5) [reply storybook/src/data-display/DataTable.mdx:131]: No, you don't need to explicitly define multiple columns in the defaultSortBy prop when you have enableMultiSort enabled for the DataTable. The defaultSortBy prop is used to set the initial sorting configuration when the table loads. If you've set canSort to true for the desired columns and enabled multi-column sorting, users can dynamically sort the table by clicking on the column headers.

File: libs/data-display/src/DataTable.tsx [hunk 2]
```diff
@@ -126,6 +126,12 @@ export type DataTableProps<T extends object> = {
    * Content to be displayed when the dataset is empty.
    */
   emptyState?: React.ReactNode;
+
+  /**
+   * Enable or disable multi-column sorting.
+   * When set to true, users can sort by multiple columns simultaneously.
+   */
+  enableMultiSort?: boolean;
```
Comment (jtomic-croz) [inline libs/data-display/src/DataTable.tsx:134]: can you rename this to multisort / multiSort? "enable" is a bit redundant for a boolean prop

**Review by jtomic-croz — CHANGES_REQUESTED** [review]

**Review by jtomic-croz — DISMISSED** [review]

**Comment by jtomic-croz** [comment]
@ivam5 Please resolve conflicts and we can merge this

**Review by jtomic-croz — APPROVED** [review]

**Pull request merged**